docs/cluster-spec.md

```diff
@@ -31,11 +31,12 @@ The following kinds are defined:
 
 For `kind='service'`, there is an additional `service` property with:
 
+* `buildType` -- the type of service build (see below)
 * `dockerImage` -- (generated by build process) the docker image containing this service
 
-## `ui`
+For `buildType == "heroku-buildpack"`, there are additional properties `stack` (the Heroku stack on which the image is based) and `buildpack` (the URL of the buildpack repository).
 
-This is a repository that will become the taskcluster web UI.
+The tools and docs sites are special-cased with `buildType == "tools-ui"` `buildType == "docs"`, which has the additional property `node` giving the node version.
```

Comment:
is there are reason these are different from `heroku-buildpack`, ie. could just run a static node server inside the dyno?
I'm guessing it has to do with the fact that they want different input, such as all the docs tarballs, is that correct?

Couldn't we make just one called `frontend` or something?

Also it's possible that these should have `kind != 'service'`, as they aren't really services...

Reply:
The two could probably be combined -- it remains to be seen how their build processes turn out, really.  Buildpacks don't allow injecting content, that's correct.

They *are* services in the sense that they start with a repo and end up with a docker image that is built to be run in Kubernetes.  That the build process is a little different is merely an implementation detail, from a deployment perspective.

It's all pretty loose for the moment until we have everything in place -- then we can revisit some of these decisions.